### PR TITLE
[DS-405]: [Pyxis React] Add `forwardRef` to Control components

### PR DESCRIPTION
**Pull request by lauratrapani**
https://prima-assicurazioni-spa.myjetbrains.com/youtrack/issue/DS-405

## 💥 Breaking Changes
The type of `onChange` prop of `RadioGroup` and `RadioCardGroup` has changed:
```js
// before
onChange: (values: string[]) => void;

// now
onChange: (value: string) => void;
```

## ✨ Enhancements
`RadioGroup.Option`,`RadioCardGroup.Option`, `CheckboxGroup.Option` and `CheckboxCardGroup.Option` now can receive a `ref` and also they can have a `onChange` prop of type:
```js
onChange: (event: ChangeEvent<HTMLInputElement>) => void;
```

### CODE REVIEW

File: packages/pyxis-react/src/components/RadioGroup/RadioGroup.tsx [hunk 1]
```diff
@@ -4,27 +4,34 @@ import { OptionLayout } from 'internalComponents/ControlGroup/ControlGroup';
...
-    layout={layout}
-    type="radio"
-  >
-    {children}
-  </ControlGroup>
-);
+  children, initialValue, layout = 'horizontal', onChange, ...props
+}) => {
+  const handleOnChange = ([value]: string[]) => {
+    if (onChange) onChange(value);
```
Comment (brianvanburken) [inline packages/pyxis-react/src/components/RadioGroup/RadioGroup.tsx:10]:
```suggestion
    onChange?.(value);
```

File: packages/pyxis-react/src/internalComponents/ControlGroup/ControlGroup.tsx [hunk 2]
```diff
@@ -72,15 +72,20 @@ const ControlGroup: FC<ControlGroupProps> = ({
       {Children.map(children,
         (child) => cloneElement(child, {
           checked: values.includes(child.props.value),
-          onChange: handleChange,
+          onChange: (event: ChangeEvent<HTMLInputElement>) => {
+            handleChange(event);
+            if (child.props.onChange) child.props.onChange(event);
```
Comment (brianvanburken) [inline packages/pyxis-react/src/internalComponents/ControlGroup/ControlGroup.tsx:77]:
```suggestion
            child.props.onChange?.(event);
```

File: packages/pyxis-react/src/components/RadioCardGroup/RadioCardGroup.tsx [hunk 3]
```diff
@@ -4,28 +4,39 @@ import { CardOptionLayout } from 'internalComponents/ControlGroup/ControlGroup';
...
-  </ControlGroup>
-);
+  children,
+  initialValue,
+  layout = 'default',
+  onChange,
+  ...props
+}) => {
+  const handleOnChange = ([value]: string[]) => {
+    if (onChange) onChange(value);
```
Comment (brianvanburken) [inline packages/pyxis-react/src/components/RadioCardGroup/RadioCardGroup.tsx:14]:
```suggestion
    onChange?.(value);
```

**Review by brianvanburken — COMMENTED** [review]
This is a different way which is shorter and achieves the same, but it is more style preference I guess :)

**Review by brianvanburken — APPROVED** [review]

**Review by emanuelemacri — APPROVED** [review]

**Review by Gsciarra — APPROVED** [review]

**Pull request merged**